daiquiri/formatter.py

```diff
@@ -23,19 +31,124 @@
 )
 
 
-class ColorFormatter(logging.Formatter):
-    # TODO(jd) Allow configuration
-    LEVEL_COLORS = {
-        logging.DEBUG: '\033[00;32m',  # GREEN
-        logging.INFO: '\033[00;36m',  # CYAN
-        logging.WARN: '\033[01;33m',  # BOLD YELLOW
-        logging.ERROR: '\033[01;31m',  # BOLD RED
-        logging.CRITICAL: '\033[01;31m',  # BOLD RED
-    }
+class Palette(object):
+    """Object that represents TTY colors."""
+
+    PREFIX = '\033['
+    BASE_WHITE = 15
+    BASE_MOD = '00;%dm'
+    CHART_WHITE = 231
+    CHART_MOD = '38;5;%dm'
+
+    def __init__(self, *args, base=False, name=None):
+        """Create a new color palette from console values.
+
+        :param args: a color value for each of the default logging levels.
+        :param base: (bool) use TTY defined colors instead of 256-color palette.
+        """
+        self.name = name
+        self.base = base
+        self.colors = list(args)
+        self._level_colors = None
+
+    def __str__(self):
+        colors = [c[1] if not isinstance(c, int) else c for c in self.colors]
```

Comment:
Use `()` instead of `[]` so it's just an iterator, no need to create a full list.